src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/BuildHostProcessManager.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
+
+internal sealed class BuildHostProcessManager : IAsyncDisposable
+{
+    private readonly string? _binaryLogPath;
+
+    private readonly SemaphoreSlim _gate = new(initialCount: 1);
+    private readonly Dictionary<BuildHostProcessKind, BuildHostProcess> _processes = new();
```

Comment:
Ah interesting - so eventually we could even have different .net core build hosts at the same time if we needed (for different projects)?

Reply:
Yep! DevKit doesn't support it, and it's not clear we should either from a user-scenario standpoint, but once we have MSBuildLocator behavior changed it's easier to write it right than maintain the current behavior. This type will eventually move to MSBuildWorkspace too where people might be doing programmatic things across multiple repositories too, so since we're an API it's harder to say "we don't support it".